src/ViewRendererBundle/objects/helper/TPkgViewRendererLessCompiler.php

```diff
@@ -23,15 +23,22 @@ class TPkgViewRendererLessCompiler
      * @var string
      */
     private $cssDir;
+
     /**
      * @var string
      */
     private $resourceCollectionRefreshPrefix;
 
-    public function __construct(string $cssDirRelativeToWebRoot, string $resourceCollectionRefreshPrefix)
+    /**
+     * @var array
+     */
+    private $additionalVariables;
+
+    public function __construct(string $cssDirRelativeToWebRoot, string $resourceCollectionRefreshPrefix, array $additionalVariables)
```

Comment:
We could think about adding a method `addLessVariable(string $key, string $value)` instead. Although constructor injection in general is preferred, we would gain a few advantages:
- In projects or third-party bundles additional variables can be defined through compiler passes.
- Users of the class don't need to learn how the array needs to look like; they can pass key and value as separate method arguments. Also type-safety is improved a bit.
- No BC break, so we could think about adding this feature to 6.2.x (only if "necessary" - as this is a feature, I'd prefer master).

Reply:
Now using a method addAdditionalVariables().